NDTensors/ext/NDTensorsCUDAExt/linearalgebra.jl

```diff
@@ -1,13 +1,39 @@
-function NDTensors.svd_catch_error(A::CuMatrix; alg="JacobiAlgorithm")
+function NDTensors.svd_catch_error(A::CuMatrix; alg::String="JacobiAlgorithm")
   if alg == "JacobiAlgorithm"
     alg = CUDA.CUSOLVER.JacobiAlgorithm()
   else
     alg = CUDA.CUSOLVER.QRAlgorithm()
   end
+  return NDTensors.svd_catch_error(A, alg)
+end
+
+function NDTensors.svd_catch_error(A::CuMatrix, ::CUDA.CUSOLVER.JacobiAlgorithm)
   USV = try
-    svd(expose(A); alg=alg)
+    svd(A; alg=CUDA.CUSOLVER.JacobiAlgorithm())
   catch
     return nothing
   end
   return USV
 end
+
+function NDTensors.svd_catch_error(A::CuMatrix, ::CUDA.CUSOLVER.QRAlgorithm)
+  s = size(A)
+  if s[1] < s[2]
+    At = copy(Adjoint(A))
+
+    USV = try
+      svd(At; alg=CUDA.CUSOLVER.QRAlgorithm())
+    catch
+      return nothing
+    end
+    MV, MS, MU = USV
+    USV = SVD(copy(MU), MS, Adjoint(MV))
```

Comment:
What's the reason for `copy(MU)` here?

Reply:
MU here is `typeof(MU) = LinearAlgebra.Adjoint{Float64, CuArray{Float64, 2, CUDA.Mem.DeviceBuffer}}`
In order to form the correct U matrix I need to force MU to evaluate the adjoint so I am doing that using copy.

Reply:
I see, so it is an issue with the `SVD` wrapper? Seems like a bad design for the `SVD` wrapper to not allow more general types.

Reply:
Do we even need the `SVD` wrapper or can we just use something like:
```julia
SVD = (MU, MS, MV')
```
or
```julia
SVD = (; U=MU, S=MS, V=MV')
```

Reply:
Good question, I am not sure I will check. Thanks!